### ISSUE TITLE
Installer closing after asking for the name of the game

### ISSUE THREAD

**Issue by XploitOcelot** [Darthagnon/FXWebPlayer-w2i-Installer#4]
Hey, thanks for your tool, on the first place.
I really wanted to give it a try, but every time I launch the .bat, just after the second question, the CMD window closes itself.

Any idea?

**Comment by Darthagnon** [comment]
@XploitOcelot 2nd Question - that's "Name of the Game", right? Immediately after that, the script generates a random number, then should display a confirmation prompt before installation. To proceed with the installation, you need the ```gamename.w2i```, ```install_w2i.bat``` and ```res``` folder next to each other. 

What's the game you're trying to install? I know Runaway has problems, because additional DRM was included in the w2i file (fixable by repacking with 7zip).

**Comment by XploitOcelot** [comment]
> 
> 
> @XploitOcelot 2nd Question - that's "Name of the Game", right? Immediately after that, the script generates a random number, then should display a confirmation prompt before installation. To proceed with the installation, you need the `gamename.w2i`, `install_w2i.bat` and `res` folder next to each other.
> 
> What's the game you're trying to install? I know Runaway has problems, because additional DRM was included in the w2i file (fixable by repacking with 7zip).

Hi! Sorry for the late answer,
at first I tried with Runaway, yes, but after I checked with two more games and the problem persists. That is, Hollywood Monsters and Imperivm.
I don't get any confirmation prompt. I just write the name of the game, push enter, and the window closes itself. That's the end of it :-/
I put both, the *.w2i file and the res folder with the batch file alltogether. Does the w2i file need to have an especific name, maybe??

EDIT: Just checked if renaming the w2i file worked, aaand no success (using Imperivm's w2i file)

**Comment by Darthagnon** [comment]
Suggestions:
1 - see if the w2i files open with 7zip. If they do, then they'll work with this script. If not, then they won't. If I remember, Runaway w2i had additional encryption that I couldn't figure out. Easy workaround was to wait for the FX store to extract it, then repack the files using 7zip and rename the extension to w2i. See #3 

2 - If you're confident with code, you can try adding ```pause``` at a few points in the script, around the stage where it crashes, and see what the console says. 

3 - I own some of the Imperivm games, and didn't have any trouble. I had trouble with Runaway. I don't own Hollywood Monsters. Would you like to share Imperivm or Hollywood Monsters install files with me so I can test? @XploitOcelot [Ping me on Telegram](https://t.me/Darthagnon) or send me an email at darren.delima[at]yahoo.co.uk. It's possible that the FX store had more games with the same DRM as Runaway, or that they've started adding DRM/encryption to the install files. I believe the FX store website was down for a while recently. Don't know what's up with FX as a company.

I should update the README with a list of games that I've successfully tested it on.

**Comment by XploitOcelot** [comment]
Hi there,
1. Yeah, from my FX digital collection, the only ones not opening with 7zip are Runaway and Imperivm III. All the others just open fine. Nevertheless, I had no success installing any game with your script for now :-(
Once I figure out why it closes before starting any installation process, I'll check my custom w2i files for the games I mentioned.
2. I'm afraid I'm not. I'm more of a betatester when trying to help with github projects, finding something not working here and there...
3. I can test any of my 16 titles if you want. No problem with that so far if we get to know why it works on some computers and not in others... I'll get to you via email!
About FX as a company... I really thought they were gonners like three or two years ago. Can't believe they are still holding up. They really should change their business model (if they live long anough to think about it).

**Comment by Darthagnon** [comment]
Right, sounds like there's additional DRM on Imperivm III, too. Be careful re-downloading the games, as the FX store only allows you a fixed number of downloads per game (can't remember the exact number; it varies, 5 or 6?)

If I were you, I'd keep the original files, but also redownload e.g. Imperivm III, wait for the legit installation process to almost finish, and while the "confirm installation" prompt is still open, check your temp files - there will be the w2i image, and a folder next to it with the extracted files. 7zip the folder with the extracted files, and change the extension to ```w2i```, so you at least have a DRM-free backup that *should* install. 

I'll re-test the script, and once you contact me, share my files with you. (you don't need to redownload Runaway; I've got a DRM-free w2i made as describe above that I'll send you)
